tests/subsys/net/lib/nrf_cloud/src/main.c

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <unity.h>
+#include <zephyr.h>
+#include <stdio.h>
+#include <net/nrf_cloud.h>
+#include <mock_nrf_cloud_transport.h>
+#include "nrf_cloud_fsm.h"
+
+/* This is required since unity_main return int and
+ * zephyr expects main not to return any value
+ */
+extern int unity_main(void);
+
+/* Simple event handler needed for the init function to
+ * have a success initialization in the test below
+ */
+void event_handler(const struct nrf_cloud_evt *const evt)
+{
+	switch (evt->type) {
+	case NRF_CLOUD_EVT_READY:
+		printk("Device is connected and ready");
+		break;
+	case NRF_CLOUD_EVT_ERROR:
+		printk("Failed to connect to cloud");
+		break;
+	default:
+		break;
+	}
+}
+
+/* This function runs before each test */
+void setUp(void)
+{
+	/* Uninit to change the internal state to idle */
+	__wrap_nct_uninit_Ignore();
+	nrf_cloud_uninit();
+}
```

Comment:
I suggest only do this in `tearDown`

Your `nfsm_get_current_state() == STATE_IDLE` assertions have you covered here